salt/modules/file.py

```diff
@@ -3818,7 +3818,7 @@ def _get_local_file_source_sum(path):
                 # but `cached_sum == source_sum['hsum']` is elliptical as prev if
                 sfn = cached_dest
                 source_sum = {'hsum': cached_sum, 'hash_type': htype}
-            elif cached_sum != source_sum['hsum']:
+            elif cached_sum != source_sum.get('hsum', __opts__.get('hash_type', 'md5')):
```

Comment:
We should not be using ``__opts__.get()`` here, nor falling back to md5, for reasons I explain further in [this pull request](https://github.com/saltstack/salt/pull/40552), which I have just opened to fix similar references in the 2016.3 branch.

Please change this to ``__opts__['hash_type']``